### PR TITLE
Add utils/format script.

### PR DESCRIPTION
**Pull request by dj2**
This is a simple script to run clang-format over all the bits of the
repo. Makes it easier when submitting a CL to get everything formatted
as neede.

### CODE REVIEW

File: utils/format [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+find examples -name "*.h"  -exec clang-format -i {} \;
+find examples -name "*.cc"  -exec clang-format -i {} \;
```
Comment (dneto0) [inline utils/format]: SPIRV-Tools uses .cpp and .hpp files.

File: utils/format [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+find examples -name "*.h"  -exec clang-format -i {} \;
```
Comment (paulthomson) [inline utils/format]: I believe clang-format can accept a list of files, which might be better, especially for those on Windows, where process creation can be slow. Perhaps not a big deal though.

File: utils/format (hunk 1 above)
Comment (dneto0) [reply utils/format]: I'd probably do a double loop

 for dir in source test tools include; do
  . ..
 ...

Comment (dneto0) [reply utils/format]: or do something like this.

alias cgrep='find . -type f \( -name "*.c" -o -name "*.cc" -o -name "*.h" -o -name "*.hpp" -o -name "*.cpp" \) -print0 | xargs -0 grep'

Comment (dj2) [reply utils/format]: done

File: utils/format (hunk 2 above)
Comment (dj2) [reply utils/format]: This seems like the easiest way to get the list of file names? I'd need to do something to loop over all dirs to find the files, yea?

**Comment by dj2** [comment]
Closing this, it seems like git clang-format master would do what we want.

**Pull request closed without merging**